src/api/wrappers/cf_wrapper.py

```diff
@@ -0,0 +1,66 @@
+"""A wrapper class for the CurseForge API.
+"""
+
+from json.decoder import JSONDecodeError
+from pathlib import Path
+from urllib.parse import urljoin
+
+import requests
+from fire.core import FireError
+
+from src.api.dataclasses.cf_response_objects import (CFDataResponse,
+                                                     CFGetModResponse)
+
+CF_BASE_URL = 'https://api.curseforge.com'
+CF_API_VERSION = 'v1'
+
+
+class CurseForgeWrapper:  # pylint: disable=too-few-public-methods
+    """Wrapper for the CurseForge API."""
+
+    def __init__(self, api_key):
+        self.api_key = api_key
+
+    def get_mod(self, mod_id: int) -> CFDataResponse:
+        """
+        Return CFGetModData object containing mod metadata
+
+        Parameters:
+          mod_id (int): The Mod ID for the mod metadata to be fetched
+
+        Returns:
+          CFGetModData:
+            Object containing CFGetModResponse object or None, statusCode of API request,
+            and status containing status or error message.
+        """
+        get_mod_endpoint = 'mods'
+        request_path = Path(CF_API_VERSION).joinpath(
+            get_mod_endpoint, str(mod_id))
+        request_url = urljoin(CF_BASE_URL, str(request_path))
+
+        headers = {
+            'Accept': 'application/json',
+            'x-api-key': self.api_key
+        }
+
+        try:
+            response = requests.get(request_url, headers=headers,
+                                    timeout=10)  # 10 second timeout
+        except requests.exceptions.HTTPError as err:
```

Comment:
put all the logic in try and catch all the exception types

Reply:
Should I break up the logic in separate try blocks to isolate different major areas where things could go wrong (see the two try blocks above), or should it all go in one try block? (i.e. combine the two existing try blocks to one and expand)